### PR TITLE
[rust] upgrade dependencies

### PR DESCRIPTION
**Pull request by nbdd0121**
The following crates are upgraded without needing to change code:
* directories: 4 -> 5
* memoffset: 0.6 -> 0.9
* zerocopy: 0.5 -> 0.6
* object: 0.25 -> 0.32
* strum: 0.24 -> 0.25
* indexmap: 1 -> 2
* deser-hjson: 1 -> 2
* mio: 0.7 -> 0.8
* mio-signal: 0.1 -> 0.2
* cryptoki-sys: 0.1.4 -> 0.1.6

The following crates are upgraded with small code changes needed:
* rsub: 0.8 -> 0.9
* rsa: 0.8 -> 0.9
* num_enum: 0.5 -> 0.7
* indicatif: 0.16 -> 0.17

It's recommended to review this PR commit by commit.

### CODE REVIEW

File: third_party/rust/Cargo.toml [hunk 1]
```diff
@@ -34,14 +34,14 @@ humantime-serde = "1.1"
 indicatif = "0.16.2"
 is-terminal = "0.4"
 log = "0.4"
-memoffset = "0.6.0"
-mio = { version = "0.7.0", features = ["os-poll", "net", "os-ext"] }
-mio-signals = "0.1.5" # mio-signals 0.2.0 exists, but requires mio 0.8.0
```
Comment (jwnrt) [inline third_party/rust/Cargo.toml:39]: Do you know why we couldn't upgrade to `0.8.0` when this comment was written?

**Review by jwnrt — APPROVED** [review]
LGTM, thanks @nbdd0121.

Was there a motivating bug for these updates, or just keeping on top of new versions?

File: third_party/rust/Cargo.toml (hunk 1 above)
Comment (nbdd0121) [reply third_party/rust/Cargo.toml:39]: I don't know. This line exists when the file is created. Mio 0.8 removes a bunch of deprecated stuff but we haven't been using the deprecated ones anyway.

Comment (jwnrt) [reply third_party/rust/Cargo.toml:39]: Fair enough, should be fine.

File: sw/host/opentitanlib/src/app/mod.rs [hunk 2]
```diff
@@ -93,7 +101,7 @@ impl ProgressIndicator for StagedProgressBar {
     fn progress(&self, pos: usize) {
         let bar = self.current_progress_bar.borrow();
         let bar = bar.as_ref().unwrap();
-        if pos as u64 == bar.length() {
+        if pos as u64 == bar.length().unwrap() {
```
Comment (HU90m) [inline sw/host/opentitanlib/src/app/mod.rs:104]: Are the bar.length unwraps safe? When is [`BarState.len`](https://docs.rs/indicatif/0.17.6/src/indicatif/state.rs.html#25) None?

**Review by HU90m — APPROVED** [review]
Looks good

File: sw/host/opentitanlib/src/app/mod.rs (hunk 2 above)
Comment (nbdd0121) [reply sw/host/opentitanlib/src/app/mod.rs:104]: It's none when a spinner is created, but we create it with `ProgressBar::new` so this should always return a value.

**Pull request merged**